src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -260,14 +260,53 @@ end = struct
     let t = {ledger; scan_state} in
     let%bind () =
       Statement_scanner_with_proofs.check_invariants scan_state
-        ~error_prefix:"Staged_ledger.of_scan_state_and_ledger" ledger
+        ~error_prefix:"Staged_ledger.of_scan_state_and_ledger"
+        (Frozen_ledger_hash.of_ledger_hash (Ledger.merkle_root ledger))
         (Some snarked_ledger_hash)
     in
```

Comment:
Could you label these ledger hash arguments so that it is a bit more clear? Something like `ledger_hash_begin` and `ledger_hash_end` ?